akka-actor/src/main/scala/akka/io/dns/DnsResourceRecords.scala

```diff
@@ -15,28 +15,28 @@ import akka.util.{ ByteIterator, ByteString, unused }
 import scala.annotation.switch
 import scala.concurrent.duration._
 
-sealed abstract class ResourceRecord(val name: String, val ttl: Ttl, val recType: Short, val recClass: Short)
+sealed abstract class ResourceRecord(val name: String, val cachePolicy: CachePolicy, val recType: Short, val recClass: Short)
```

Comment:
I like this better because it seems to align with the actual protocol better, but I may be missing some consequence of it.

Reply:
On the one hand it might follow the semantics of the RFC standard less strictly, because the RFC states that a TTL is a 32 bit integer. On the other hand it allows to model the interpretation of the TTL as documented in 3.2.1:
> Zero values are interpreted to mean that the RR can only be used for the transaction in progress, and should not be cached.
However, it also allows a RR with a `Forever` TTL and I'm not sure if that's desired.

Reply:
I wouldn't like to do such a big change, again. This API has been moving so much and we just marked it as stable. Can't we solve it in a way that is backwards compatible, like just accepting that Ttl can be zero?

Reply:
Ok, let's do that, sorry for leading you in a different direction @agboom

Reply:
to clarify: `fromPositive` should still not accept zero (would be weird naming), if needed we can add a new factory method for the zero case `Ttl.never`

Reply:
Ok, I'll revert the previous commit and implement as you suggested. No problem @johanandren, that's how I've learned something :)